reactive-banana/src/Reactive/Banana/Combinators.hs

```diff
@@ -110,20 +111,23 @@ never = E Prim.never
 -- >    | timex >  timey = (timey,y)     : unionWith f ((timex,x):xs) ys
 -- >    | timex == timey = (timex,f x y) : unionWith f xs ys
 unionWith :: (a -> a -> a) -> Event a -> Event a -> Event a
-unionWith f = mergeWith Just Just (\x y -> Just (f x y))
+unionWith f = mergeWith id id f
+
+-- | Merge two event streams of any type.
+merge :: Event a -> Event b -> Event (These a b)
+merge e1 e2 = E $ Prim.mergeWith This That These (unE e1) (unE e2)
 
 -- | Merge two event streams of any type.
 --
--- This function generalizes 'unionWith', and can be used to filter event
--- occurrences as well.
+-- This function generalizes 'unionWith'.
 mergeWith
-  :: (a -> Maybe c) -- ^ The function called when only the first event emits a value.
-  -> (b -> Maybe c) -- ^ The function called when only the second event emits a value.
-  -> (a -> b -> Maybe c) -- ^ The function called when both events emit values simultaneously.
+  :: (a -> c) -- ^ The function called when only the first event emits a value.
+  -> (b -> c) -- ^ The function called when only the second event emits a value.
+  -> (a -> b -> c) -- ^ The function called when both events emit values simultaneously.
   -> Event a
   -> Event b
   -> Event c
-mergeWith f g h e1 e2 = E $ Prim.mergeWith f g h (unE e1) (unE e2)
+mergeWith f g h e1 e2 = these f g h <$> merge e1 e2
```

Comment:
Why not define it in terms of Prim.mergeWith? Just curious.

Reply:
Then merge can be just
merge = mergeWith This That These

Reply:
Sounds good! I'll make that change